src/argilla_sdk/records/_resource.py

```diff
@@ -273,15 +273,18 @@ def to_dict(self) -> Dict[str, Dict]:
             `{"fields": {"prompt": "...", "response": "..."}, "responses": {"rating": "..."},
         """
         fields = self.fields.to_dict()
-        metadata = self.metadata
+        metadata = self.metadata.to_dict()
```

Comment:
Oh Since metadata is now an extension of [dict class](https://github.com/argilla-io/argilla-python/pull/204),  I removed this method on [this commit](https://github.com/argilla-io/argilla-python/pull/197/commits/61e5b7aeceeda86b276d9696cb656d7ba12f2359) (also because it wasn't used)

Let me know if I should revert the commit, please.

Reply:
It either needs to be a dict or have a to_dict method to be serialized. This is its only reference, so I can deal with it on merging.